libs/clients/ship-registry/src/lib/ship-registry.service.ts

```diff
@@ -0,0 +1,96 @@
+import { Injectable } from '@nestjs/common'
+import { FindShipByNameOrNumberRequest, ShipApi } from '../../gen/fetch'
+import { FetchError } from '@island.is/clients/middlewares'
+
+interface Ship {
+  skraningarnumer: number
+  umdaemisnafn: string
+  heimahofn: string
+  umdaemistolustafir: number | null
+  bruttorumlestir: number
+  bruttoTonn: number
+  smidaar: number
+  nafnskips: string
+  skraningarstada: string
+  skraningarlengd: number
+  umdaemisbokstafir: string
+  gerd: string
+  smidastod: string
+  eigendur: {
+    kennitala: string
+    nafn: string
+    eignaprosenta: number
+  }[]
+  engines: {
+    manufacturingNumber: null
+    power: number
+    year: number
+    usage: {
+      name: string
+    }
+    manufacturer: {
+      code: string
+      name: string
+    }
+  }[]
+  fishery: {
+    identityNumber: string
+    name: string
+    address: string
+    postalCode: string
+  }
+}
+
+const getValueOrEmptyString = (value?: string | number | null) => {
+  return value ? value : ''
+}
+
+const mapRegion = (ship: Ship) => {
+  return `${getValueOrEmptyString(ship.umdaemisbokstafir)}${
```

Comment:
This method is not very readable. Definitely consider to split up, store in variables, etc.